src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2195,27 +2195,36 @@ private String getSignatureForInstanceTTEFn(
             + classTemplatizedType
             + " ;";
       }
-      return null;
+      if (propName.equals("when")) {
+        return "when < RESULT, T > (value: T, successCallback: (promiseValue: T) => "
+            + classTemplatizedType
+            + "|RESULT, errorCallback: null | undefined | "
+            + " ((reason: any) => any), notifyCallback?: (state: any) => any): "
+            + classTemplatizedType
+            + ";";
+      }
+      String promiseName = className + ".Promise";
+      return getPromiseMethod(propName, promiseName);
     }
 
     private String getSignatureForStaticTTEFn(String propName, FunctionType ftype) {
       // If ftype is foo.bar.Promise.all, extract className as Promise.
       String className = getUnqualifiedName(getNamespace(ftype.getDisplayName()));
+      String maybePromiseMethod = getPromiseMethod(propName, className);
+      if (maybePromiseMethod != null) {
+        return "static " + maybePromiseMethod;
+      }
+      return null;
+    }
+
+    private String getPromiseMethod(String propName, String className) {
       switch (propName) {
         case "resolve":
-          return "static resolve < T >(value: "
-              + className
-              + " < T > | T): "
-              + className
-              + " < T >;";
+          return "resolve < T >(value: " + className + " < T > | T): " + className + " < T >;";
         case "race":
-          return "static race < T > (values : T [] ) : " + className + " < T > ;";
+          return "race < T > (values : T [] ) : " + className + " < T > ;";
         case "all":
-          return "static all(promises : "
-              + className
-              + " < any > [] ) : "
-              + className
-              + " < any [] > ;";
+          return "all(promises : " + className + " < any > [] ) : " + className + " < any [] > ;";
```

Comment:
Ended up not so simple, because .all has to support also tuple types. If we are to do this correctly we have to emit *all* overrides for all from here:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/angular/index.d.ts#L1014

I am leaving this for another PR.